mathesar_ui/src/component-library/toast/ToastPresenter.svelte

```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+  import type { SvelteComponent } from 'svelte';
+  import { fade, fly } from 'svelte/transition';
+  import { flip } from 'svelte/animate';
+  import type { Readable } from 'svelte/store';
+  import type { ToastEntry } from './ToastController';
+  import ToastItem from './ToastItem.svelte';
+
+  export let entries: Readable<ToastEntry[]>;
+  export let toastItemComponent: typeof SvelteComponent = ToastItem;
+</script>
+  
+<ul class="toast-presenter">
+  {#each $entries as entry (entry.id)}
+    <li in:fly={{ x: 256 }} out:fade animate:flip={{ duration: 200 }}>
+      <svelte:component this={toastItemComponent} {entry} />
```

Comment:
I'm okay with passing down a component instead of slot.

Users may want to provide additional props for the custom component. It might be better to pass them down with:
 ```suggestion
       <svelte:component this={toastItemComponent} {...$$restProps} {entry}/>
```

restProps isn't recommended unless we do not know what the props could be, which is true in this case.

Reply:
Addressed in 1035b68b9d52ac22e47d6070e3dc8e8acbe30628